api/v1alpha1/runnerpool_types.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: Some lines of the code are proudly copied from Kubernetes project.
+// https://github.com/kubernetes/apimachinery/tree/master/pkg/apis/meta/v1
+// https://github.com/kubernetes/api/tree/master/core/v1
+// https://github.com/kubernetes/api/tree/master/apps/v1
+
+// RunnerPoolSpec defines the desired state of RunnerPool
+type RunnerPoolSpec struct {
+	// RepositoryName describes repository name to register Pods as self-hosted
+	// runners.
+	RepositoryName string `json:"repositoryName"`
+
+	// Number of desired pods. This is a pointer to distinguish between explicit
+	// zero and not specified. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Label selector for pods. Existing ReplicaSets whose pods are
+	// selected by this will be the ones affected by this deployment.
+	// It must match the pod template's labels.
+	Selector *metav1.LabelSelector `json:"selector"`
+
+	// Template describes the pods that will be created.
+	Template PodTemplateSpec `json:"template"`
```

Comment:
Is this field required?
If it is empty, can't I use the default value?

Reply:
Yes, I basically agree with you but setting default manifest requires not a little cost but if we do it later with less cost, say, after running for a few months.
I keep it simple for now and wanna do it after I confirm that this controller stably runs.